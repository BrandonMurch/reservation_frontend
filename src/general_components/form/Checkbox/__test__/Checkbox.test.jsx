import React from 'react';
import { render, fireEvent, getByTestId } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import CheckboxAndLabel from '../index';

describe('<CheckboxAndLabel />', () => {
  let component;
  let mockFunction;
  let container = null;
  beforeEach(() => {
    mockFunction = jest.fn();
    component = render(<CheckboxAndLabel
      label="checkbox label"
      name="checkbox name"
      onClick={mockFunction}
    />);
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  it('should match snapshot', () => {
    const tree = renderer.create(<CheckboxAndLabel
      label="checkbox label"
      name="checkbox name"
      onClick={mockFunction}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should display the checkbox label', () => {
    const element = component.getByText(/checkbox label/i);
    expect(element).toBeInTheDocument();
  });
  it('should perform the function on click', () => {
    const element = getByTestId(component.container, 'checkbox');
    fireEvent.click(element);
    expect(mockFunction).toHaveBeenCalled();
  });
});
