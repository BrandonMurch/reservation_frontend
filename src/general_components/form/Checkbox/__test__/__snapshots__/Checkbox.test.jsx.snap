// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxAndLabel /> should match snapshot 1`] = `
<div
  className="inputGroup"
>
  <label
    className="labelText"
    htmlFor="checkbox name"
  >
    checkbox label:
  </label>
  <input
    data-testid="checkbox"
    name="checkbox name"
    onClick={[Function]}
    required={true}
    type="checkbox"
    value={false}
  />
</div>
`;
